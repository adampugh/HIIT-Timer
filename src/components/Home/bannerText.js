import React from "react";

const bannerText = () => (
    <div className="bannerText">
        <div className="bannerText__container">
            <h2>
                Create custom high intentsity interval workouts with HIIT Timer. Get in shape and loose weight with HIIT Timer today! 
            </h2>
        </div>
    </div>
);

export default bannerText;