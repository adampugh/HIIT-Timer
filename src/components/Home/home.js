import React from "react";
import Navbar from "../UI/navbar";
import VideoBanner from "./videoBanner";
import TextBanner from "./bannerText";
import PhoneBanner from "./phoneBanner";
import Panels from "./panels";
import Footer from "../UI/footer";


const home = () => (
    <div>
        <Navbar links={["Sign Up", "Log In"]} classes={"navbar--loggedOut"} />
        <VideoBanner />
        <TextBanner />
        <PhoneBanner />
        <Panels />
        <Footer />
    </div>
);

export default home;