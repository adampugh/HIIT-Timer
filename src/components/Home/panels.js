import React from "react";
import armIcon from "../../assets/photos/arm-icon.png";
import dumbbellIcon from "../../assets/photos/dumbbell-icon.png";
import pencilIcon from "../../assets/photos/pencil-icon.png";


const panels = () => {
    return (
        <div className="homecontent__wrapper">
            <div className="container homecontent">
                <h2>Create workouts today for fantasic that are actually really fun.</h2>
                <div className="grid--3to1 homecontent__grid">
                    <div className="homecontent__panel">
                        <img src={armIcon} alt="Arm icon" />
                        <h3>Create Workouts</h3>
                        <p>This is some text that is good but might not be. Don't ask me. Where Black Friday comsumers try to spend and where are those orders coming from.</p>
                    </div>
                    <div className="homecontent__panel">
                        <img src={dumbbellIcon} alt="Dumbbell icon" />
                        <h3>Customize Workouts</h3>
                        <p>This is some text that is good but might not be. Don't ask me. Where Black Friday comsumers try to spend and where are those orders coming from.</p>
                    </div>
                    <div className="homecontent__panel">
                        <img src={pencilIcon} alt="Pencil icon" />
                        <h3>Edit Workout</h3>
                        <p>This is some text that is good but might not be. Don't ask me. Where Black Friday comsumers try to spend and where are those orders coming from.</p>
                    </div>
                </div>
            </div>
        </div>
    )
}
export default panels;