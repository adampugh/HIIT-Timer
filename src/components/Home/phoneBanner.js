import React from "react";
import PhoneImage from "../../assets/photos/iphone.png"

const phoneBanner = (props) => (
    <div className="phoneBanner">
        <div className="container">
            <div className="phoneBanner__grid">
                <img src={PhoneImage} alt="phone" />
                <div className="phoneBanner__grid__panel">
                    <h3>High-intensity interval-training (HIIT) workouts are designed to build muscle and burn fat in a short period of time. Research has shown that HIIT regiments have successfully produced significant reductions in the fat mass of the whole-body.</h3>
                    <h3>HIIT Timer allows you to get a full body cardiovascular workout wherever you are. Workout using your mobile, tablet or laptop.</h3>
                    <button 
                        className="btn--blue"
                        onClick={props.handleOpenModal}
                        >Sign Up</button>
                </div>
            </div>
        </div>
    </div>
);

export default phoneBanner;