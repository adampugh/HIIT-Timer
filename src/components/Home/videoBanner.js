import React from "react";

import workoutPicture from "../../assets/photos/Workout.jpg";
import videoMP4 from "../../assets/videos/Workout.mp4";
import videoWebM from "../../assets/videos/Workout.webm";


const videoBanner = () => {
    return (

        <header id="header">
            <div className="video__container">
                <div className="video__overlay">
                    <h1>HIIT Timer</h1>
                    <p>Sign up to our weekly publication.</p>
                    <button className="btn">Log In</button>
                </div>
                <video poster={workoutPicture} autoPlay muted loop playsInline>
                    <source src={videoMP4} type="video/mp4" /> 
                    <source src={videoWebM} type="video/webm" /> 
                </video>
            </div>
        </header>
    );
}

export default videoBanner;