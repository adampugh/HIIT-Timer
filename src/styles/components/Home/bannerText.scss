.bannerText {
    background: #f1f1f1;
    padding: 130px 0;
    text-align: center;
    color: rgb(179, 179, 179);
}


.bannerText__container {
    width: 90%;
    margin: 0 auto;
}
  
@media (min-width: $desktop-breakpoint) {
    .bannerText__container {
      width: 54%;
    }
}