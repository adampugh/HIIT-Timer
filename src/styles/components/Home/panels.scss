.homecontent__wrapper {
    background: #f1f1f1;
    padding-bottom: 80px;
}

.homecontent {
    text-align: center;
    padding-top: 1em;
    h2 {
        color: rgb(179, 179, 179);
        font-size: 1.5em;
    }
}


.homecontent__grid {
    display: grid;
    grid-template-columns: 1fr;
    margin-top: 4em;
    padding-bottom: 4em;
}

.homecontent__panel {
    align-self: center;
    background: #fcfbfb;
    margin: 1em;
    padding-top: 1em;
    border-radius: 10px;
    @include boxshadow;
    h3 {
        padding-top: 1em;
        // font-size: 1.17em;
        color: slateblue;
    }
    img {
        margin-top: 2em;
        width: 100px;
    }
    p {
        padding: 1.17em;
        margin: 0 1em;
        color: grey;
        font-size: 1.1em;
    }
}


@media only screen and (min-width: $desktop-breakpoint) {
    .homecontent__wrapper {
        padding-top: 60px;
    }

    .homecontent__grid {
        margin-top: 4em;
        grid-template-columns: 1fr 1fr 1fr;
    }
}