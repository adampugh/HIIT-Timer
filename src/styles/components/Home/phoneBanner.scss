.phoneBanner {
    background: linear-gradient(141deg, rgba(106, 91, 204, 0.6) 0%, rgba(63, 119, 223, 0.6) 51%, rgba(70, 178, 218, 0.6) 75%);
    color: white;
    padding: 80px 0;
}

.phoneBanner__grid {
    display: grid;
    grid-template-columns: 1fr;
    text-align: center;
    img {
        transform: rotate(-7deg);
        height: 550px;
        margin: 0 auto;
    }
    h2 {
        font-family: 'Montserrat', sans-serif;
        padding: 40px 0;
        font-weight: 300;
    }
}

@media (min-width: $desktop-breakpoint) {
    .phoneBanner__grid {
        grid-template-columns: 1fr 1fr;
    }
}