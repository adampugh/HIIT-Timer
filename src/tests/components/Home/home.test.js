import React from "react";
import { shallow } from "enzyme"; 

import Home from "../../../components/Home/home";

test("should render homePage correctly", () => {
    const wrapper = shallow(<Home />);
    expect(wrapper).toMatchSnapshot();
});