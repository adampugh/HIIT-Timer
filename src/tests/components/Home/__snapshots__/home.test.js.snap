// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render homePage correctly 1`] = `
<div>
  <navbarLoggedOut
    handleCloseModal={[Function]}
    handleOpenModal={[Function]}
    modalOpen={false}
  />
  <videoBanner
    handleOpenModal={[Function]}
  />
  <bannerText />
  <phoneBanner
    handleOpenModal={[Function]}
  />
  <panels />
  <footer />
</div>
`;
